File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String res = s.toLowerCase().replaceAll("[^a-z0-9]", ""); // space is included here
        System.out.println(res);
        String reversed = new StringBuilder(res).reverse().toString();
        return res.equals(reversed);
        
    }
}

File: 125-valid-palindrome/valid-palindrome.py
import re
class Solution:
    def isPalindrome(self, s: str) -> bool:
        clean = [c.lower() for c in s if c.isalnum()]
        l = 0
        r = len(clean) -1
        while l < r:
            if clean[l]!=clean[r]:
                return False
            l+=1
            r-=1
        return True

        